### DISCUSSION TITLE
Pattern Data Follow Hex Editor Location?

### DISCUSSION THREAD

**Discussion by utkonos** [WerWolv/ImHex#2037] [Get Help] [answered]
I'm not sure if this is a bug, user error, or a feature request. If I have opened a PE file in ImHex version 1.36.2 on macOS, and the PE parsing pattern has run successfully, a Pattern Data tab opens with the tree structure of the parsed file. When I click a particular location in the hex editor view, the Data Inspector tab follows that click and displays the data for the selected location. However, the Pattern Data tree does not follow the click. Is there a configuration option that I am missing which causes the Pattern Data tab content to follow the selection in the hex editor? Or is this supposed to happen automagically already and there is a bug? Or is this a feature that could be implemented in the future?

**Comment by WerWolv** [comment, marked as answer]
Hey, no this is just not implemented in the way you're expecting it to be because these trees can be rather large and always jumping to the current pattern might cause slowdowns and annoyances. Instead you can right click on a highlight in the hex editor view and select `Jump To -> Current Pattern` which will select that pattern in the pattern data view.

![image](https://github.com/user-attachments/assets/b2ac1c3e-3c33-4527-950c-233ffb25e9ce)

**Reply by utkonos** [reply]
I see it now. This is good enough. Thanks!! Would you consider a configuration checkbox somewhere that automates this, but defaults to disabled so that slowdowns and annoyances are limited to people who are ok with slowdowns and annoyances as a tradeoff for the Pattern Data tab following the hex editor? I wouldn't want it to follow based on mouseover, btw. Just on selection.

**Reply by WerWolv** [reply]
I'm open for suggestions and PRs, sure. As long as it can be done in a way that doesn't just turn the app into a stuttering mess when select-scrolling through the editor
